great_expectations/rule_based_profiler/parameter_builder/histogram_single_batch_parameter_builder.py

```diff
@@ -185,121 +131,90 @@ def _build_parameters(
 
         element: Any
         if bins is None or all(element is None for element in bins):
-            is_categorical = True
-        elif not is_categorical:
-            ndarray_is_datetime_type: bool = is_ndarray_datetime_dtype(
-                data=bins,
-                parse_strings_as_datetimes=True,
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"""Partitioning values for {self.__class__.__name__} by \
+{self._column_partition_metric_single_batch_parameter_builder_config.name} into bins encountered empty or non-existent \
+elements.
+"""
             )
-            bins_ndarray_as_float: MetricValue
-            if ndarray_is_datetime_type:
-                bins_ndarray_as_float = convert_ndarray_datetime_to_float_dtype(
-                    data=bins
-                )
-            else:
-                bins_ndarray_as_float = bins
 
-            is_categorical = ndarray_is_datetime_type or not np.all(
-                np.diff(bins_ndarray_as_float) > 0.0
+        if not np.issubdtype(bins.dtype, np.number):
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"""Partitioning values for {self.__class__.__name__} by \
+{self._column_partition_metric_single_batch_parameter_builder_config.name} did not yield bins of supported data type.
+"""
             )
 
-        fully_qualified_column_values_nonnull_count_metric_parameter_builder_name: str = f"{RAW_PARAMETER_KEY}{self._column_values_nonnull_count_metric_single_batch_parameter_builder_config.name}"
+        # Only unique "bins" are necessary (hence, "n_bins" is potentially lowered to fit data distribution).
+        bins = sorted(list(set(bins)))
```

Comment:
I think we can sort the `set` by itself.

Reply:
Squashing the bins that don't provide additional information makes sense here, although we might not want to do this when using `column.partition` in other circumstances.